homeassistant/components/swiss_public_transport/config_flow.py

```diff
@@ -63,25 +98,103 @@ async def async_step_user(
+                    self._abort_if_unique_id_configured()
                     return self.async_create_entry(
                         title=unique_id,
                         data=user_input,
                     )
 
         return self.async_show_form(
             step_id="user",
-            data_schema=DATA_SCHEMA,
+            data_schema=self.add_suggested_values_to_schema(
+                data_schema=USER_DATA_SCHEMA,
+                suggested_values=user_input,
+            ),
+            errors=errors,
+            description_placeholders=PLACEHOLDERS,
+        )
+
+    async def async_step_time_fixed(
+        self, time_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Async time step to set up the connection."""
+        return await self._async_step_time_mode(
+            CONF_TIME_FIXED, vol.Schema(ADVANCED_TIME_DATA_SCHEMA), time_input
+        )
+
+    async def async_step_time_offset(
+        self, time_offset_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Async time offset step to set up the connection."""
+        return await self._async_step_time_mode(
+            CONF_TIME_OFFSET,
+            vol.Schema(ADVANCED_TIME_OFFSET_DATA_SCHEMA),
+            time_offset_input,
+        )
+
+    async def _async_step_time_mode(
+        self,
+        step_id: str,
+        time_mode_schema: vol.Schema,
+        time_mode_input: dict[str, Any] | None = None,
+    ) -> ConfigFlowResult:
+        """Async time mode step to set up the connection."""
+        errors: dict[str, str] = {}
+        if time_mode_input is not None:
+            unique_id = unique_id_from_config({**self.user_input, **time_mode_input})
+            await self.async_set_unique_id(unique_id)
+            self._abort_if_unique_id_configured()
+
+            session = async_get_clientsession(self.hass)
+            time_offset: dict[str, int] | None = time_mode_input.get(CONF_TIME_OFFSET)
+            opendata = OpendataTransport(
+                self.user_input[CONF_START],
+                self.user_input[CONF_DESTINATION],
+                session,
+                via=self.user_input.get(CONF_VIA),
+                time=time_mode_input.get(CONF_TIME_FIXED),
+            )
+            if time_offset:
+                offset_opendata(opendata, time_offset)
+            err = await self.fetch_connections(opendata)
+            if err:
+                errors["base"] = err
+            else:
+                return self.async_create_entry(
+                    title=unique_id,
+                    data={**self.user_input, **time_mode_input},
+                )
+
+        return self.async_show_form(
+            step_id=step_id,
+            data_schema=time_mode_schema,
             errors=errors,
             description_placeholders=PLACEHOLDERS,
         )
+
+    async def fetch_connections(self, opendata: OpendataTransport) -> str | None:
```

Comment:
Maybe we can put `OpendataTransport()` also in here to now duplicate the code above (and pass in the `user_input`)